### PR TITLE
Improve localpart email validation

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review.

File: core/lib/spree/core/validators/email.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class EmailValidator < ActiveModel::EachValidator
   LOCAL_PART_MAX_LENGTH = 64
   HOSTNAME_MAX_LENGTH   = 255
-  disallowed            = /[^@\s\."]/.freeze
-  LOCAL_PART_PATTERN    = /\A#{disallowed}(?:\.?#{disallowed})*\z/i.freeze
+  allowed               = /[A-Za-z0-9_+'-]/.freeze
```
Comment (dkubb) [inline core/lib/spree/core/validators/email.rb]: Can we replace `A-Za-z0-9_` with `\w`?

File: core/spec/lib/spree/core/validators/email_spec.rb [hunk 2]
```diff
@@ -34,17 +34,26 @@ class Tester
...
+    'foo..bar@email.com',
     'invalidemailemail.com',
     '@invalid.email@email.com',
     'invalid@email@email.com',
     'invalid.email@@email.com',
     "invalid.email@a#{'a.' * 126}com",
     'invalid@a.c',
     'invalid@invalid.invalidtld',
-    # This is valid according to the RFCs but we opt out and
+    # These is valid according to the RFCs but we opt out and
```
Comment (dkubb) [inline core/spec/lib/spree/core/validators/email_spec.rb]: This should be "These are"

File: core/lib/spree/core/validators/email.rb (hunk 1 above)
Comment (mbj) [inline core/lib/spree/core/validators/email.rb]: yeah, good catch.

**Pull request merged**